app/bundles/CampaignBundle/EventListener/CampaignSubscriber.php

```diff
@@ -50,13 +60,73 @@ public function __construct(IpLookupHelper $ipLookupHelper, AuditLogModel $audit
     public static function getSubscribedEvents()
     {
         return [
-            CampaignEvents::CAMPAIGN_POST_SAVE   => ['onCampaignPostSave', 0],
-            CampaignEvents::CAMPAIGN_POST_DELETE => ['onCampaignDelete', 0],
-            CampaignEvents::CAMPAIGN_ON_BUILD    => ['onCampaignBuild', 0],
+            CampaignEvents::CAMPAIGN_POST_SAVE         => ['onCampaignPostSave', 0],
+            CampaignEvents::CAMPAIGN_POST_DELETE       => ['onCampaignDelete', 0],
+            CampaignEvents::CAMPAIGN_ON_BUILD          => ['onCampaignBuild', 0],
+            CampaignEvents::ON_CAMPAIGN_TRIGGER_ACTION => ['onCampaignTriggerAction', 1],
         ];
     }
 
     /**
+     * @param CampaignExecutionEvent $event
+     *
+     * @return CampaignExecutionEvent
+     */
+    public function onCampaignTriggerAction(CampaignExecutionEvent $event)
+    {
+        if (!$event->checkContext('campaign.remoteurl')) {
+            return;
+        }
+        $config  = $event->getConfig();
+        $timeout = 10;
+        $headers = [];
+        if (!empty($config['authorization_header'])) {
+            if (strpos($config['authorization_header'], ':') !== false) {
+                list($key, $value) = explode(':', $config['authorization_header']);
+            } else {
+                $key   = 'Authorization';
+                $value = $config['authorization_header'];
+            }
+            $headers[trim($key)] = trim($value);
+        }
+
+        try {
+            $url    = $config['url'];
+            $method = $config['method'];
+            $data   = !empty($config['additional_data']['list']) ? $config['additional_data']['list'] : '';
+            $data   = array_flip(AbstractFormFieldHelper::parseList($data));
+            if (in_array($method, ['get', 'trace'])) {
+                $response = $this->connector->$method(
+                    $url.(parse_url($url, PHP_URL_QUERY) ? '&' : '?').http_build_query($data),
+                    $headers,
+                    $timeout
+                );
+            } elseif (in_array($method, ['post', 'put', 'patch'])) {
+                $response = $this->connector->$method(
+                    $url,
+                    $data,
+                    $headers,
+                    $timeout
+                );
+            } elseif ($method == 'delete') {
+                $response = $this->connector->$method(
+                    $url,
+                    $headers,
+                    $timeout,
+                    $data
+                );
+            }
+            if (in_array($response->code, [200, 201])) {
+                return $event->setResult(true);
+            }
```

Comment:
I'd also add something like
```
else {
    $event->setFailed('Response returned HTTP code '.$response->code);
}
```
I'm not sure whether the error is run through the translator later. Maybe you'll have to add Translator as a dependency to this Subscriber.